src/OrchardCore.Modules/OrchardCore.Roles/Migrations/SystemRolesMigrations.cs

```diff
@@ -0,0 +1,37 @@
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Logging;
+using OrchardCore.Data.Migration;
+using OrchardCore.Security;
+
+namespace OrchardCore.Roles.Migrations;
+
+public sealed class SystemRolesMigrations : DataMigration
+{
+    private readonly ISystemRoleNameProvider _systemRoleNameProvider;
+    private readonly RoleManager<IRole> _roleManager;
+    private readonly ILogger _logger;
+
+    public SystemRolesMigrations(
+        ISystemRoleNameProvider systemRoleNameProvider,
+        RoleManager<IRole> roleManager,
+        ILogger<RolesMigrations> logger)
+    {
+        _systemRoleNameProvider = systemRoleNameProvider;
+        _roleManager = roleManager;
+        _logger = logger;
+    }
+
+    public async Task<int> CreateAsync()
+    {
+        var systemRoles = await _systemRoleNameProvider.GetSystemRolesAsync();
+
+        foreach (var role in systemRoles)
+        {
+            await _roleManager.CreateAsync(new Role { RoleName = role });
```

Comment:
You should check if it exists before creating it because it is going to run in sites where it already exists.